components/TimeSlotsButton/index.test.js

```diff
@@ -0,0 +1,38 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import TimeslotButton from ".";
+const timeslot = {
+  id: 1,
+  startTime: "08:00",
+  endTime: "08:50",
+  bookings: [
+    {
+      id: 1,
+      numberOfPeople: 6,
+      horses: [
+        {
+          horseId: 1,
+        },
+        {
+          horseId: 3,
+        },
+        {
+          horseId: 5,
+        },
+        {
+          horseId: 6,
+        },
+        {
+          horseId: 7,
+        },
+      ],
+    },
+  ],
+};
+
+test("toggles the form when the button is clicked", () => {
+  render(<TimeslotButton timeslot={timeslot} />);
+  const button = screen.getByText("08:00 - 08:50");
+  fireEvent.click(button);
+  expect(button).toBeInTheDocument();
+});
```

Comment:
Not quite! 😆 

The test says: _"toggles the form when the button is clicked"._
What your test does: checks if the very button that you click is displayed.

The button is displayed regardless. No matter if you click it or not, it's there. You will need to check what else is displayed after the button is clicked, which is the form. See if you can find something inside the form that you can check to be in the document after that click.

Reply:
Not sure if I understand but is this what you mean:
![Screenshot 2023-09-06 at 11 34 56](https://github.com/Charlotte-Ketteler/capstone-project/assets/127936697/d8407c74-b769-4a4a-a4a3-6ec569bfd368)

Reply:
Exactly!